app/controllers/IdentityController.scala

```diff
@@ -45,7 +48,8 @@ class IdentityController(
       .fold(
         err => {
           SafeLogger.error(scrub"Failed to set password using guest account registration token ${request.body.guestAccountRegistrationToken}: ${err.toString}")
-          InternalServerError
+          warn()
+          NotFound
```

Comment:
I'm not sure that NotFound is an appropriate status code in these cases. I guess you are changing this so that these endpoints no longer trigger the High 5XX rate alarm?

If so, perhaps updating the threshold on that alarm (to accommodate the fact that we expect these calls to fail occasionally) would be better?

Reply:
Trouble is the background is that given the average rate of contribtuions, we could be 5xx every single attempt to sign up and it would barely trigger the alarm as it is.
I certainly agree that it should be a 500 error from a technical point of view, but after long discussion with @jranks123 I came to the conclusion that 5xx itself isn't granular enough given the wide range of things the app does.  We want to treat 5xx differently depending which endpoint caused them which is not possible without separate ELBs or lambdas.

As an alternative compromise, could we increase the threshold as you suggest, and change the "warning" metric I just added to be "ErrorCount" and have a PreCheckout/PostCheckout dimension.   Then we can alarm more urgently on the former.
That does however mean if the server can't even hit cloudwatch for some reason or doesn't get through that code path, we wouldn't actually alert as quickly.

Any thoughts?

Reply:
Fair point, perhaps we can try your existing solution first and see if it helps us 👍